src/highdicom/seg/sop.py

```diff
@@ -535,6 +535,33 @@ def __init__(
             plane_orientation == source_plane_orientation
         )
 
+        if self._coordinate_system == CoordinateSystemNames.SLIDE:
+            self.ImageOrientationSlide = \
+                plane_orientation[0].ImageOrientationSlide
+            if are_spatial_locations_preserved:
+                self.TotalPixelMatrixOriginSequence = \
+                    source_images[0].TotalPixelMatrixOriginSequence
+            else:
+                plane_position_values = np.array([
+                    (
+                        float(pos[0].XOffsetInSlideCoordinateSystem),
+                        float(pos[0].YOffsetInSlideCoordinateSystem),
+                        float(pos[0].ColumnPositionInTotalImagePixelMatrix),
+                        float(pos[0].RowPositionInTotalImagePixelMatrix),
+                    )
+                    for pos in plane_positions
+                ])
```

Comment:
I don't think this can be correct. Firstly it's redundant, since `plane_position_value` is calculated a few lines down. Secondly, why do you not you not use? `DimensionIndexSequence.get_index_values()`. Thirdly, what if the first frame is omitted due to being empty, would that not make the value calculated here wrong?

Reply:
> Firstly it's redundant, since plane_position_value is calculated a few lines down.

We need to compute it here before empty frames will be removed, but you are right that we shouldn't compute it twice. I will instead compute it only once before frames are being removed and subsequently use `source_image_indices` to select the remaining subset.

> Secondly, why do you not you not use DimensionIndexSequence.get_index_values()?

You are right. I should re-use that.

> Thirdly, what if the first frame is omitted due to being empty, would that not make the value calculated here wrong?

No, that's the whole point of why I am doing this. The total pixel matrix includes the empty frames that will be omitted from the data set and we need to compute the offset of the first frame before any frames will be removed. Otherwise, the image position may be wrong.

Reply:
Addressed via e7d28243432c1ae47d12a36a8fdfdbdd84ab65ab

Reply:
cc @dclunie for awareness

Reply:
Ok I think this makes sense